core/src/main/java/co/cask/wrangler/internal/RecordConvertor.java

```diff
@@ -136,9 +136,9 @@ public Schema toSchema(String id, List<Record> records) throws RecordConvertorEx
           Schema arraySchema = null;
           try {
             arraySchema = generateJSONArraySchema(array);
-          } catch (UnsupportedTypeException e) {
+          } catch (UnsupportedOperationException | UnsupportedTypeException e) {
```

Comment:
Instead of handling UnsupportedOperationException, isn't it better to throw the appropriate exception in the first place?
https://github.com/hydrator/wrangler/pull/64/files